transformer/cnbcontainerizer.go

```diff
@@ -0,0 +1,174 @@
+	if err != nil {
+		logrus.Errorf("unable to load config for Transformer %+v into %T : %s", t.Config.Spec.Config, t.CNBConfig, err)
+		return err
+	}
+	envInfo := environment.EnvInfo{
+		Name:        tc.Name,
+		ProjectName: t.Env.GetProjectName(),
+		Source:      t.Env.GetEnvironmentSource(),
+	}
+	t.CNBEnv, err = environment.NewEnvironment(envInfo, nil, environmenttypes.Container{
+		Image:      t.CNBConfig.BuilderImageName,
+		WorkingDir: filepath.Join(string(filepath.Separator), "tmp"),
+	})
+	if err != nil {
+		if !container.IsDisabled() {
+			logrus.Errorf("Unable to create CNB environment : %s", err)
+			return err
+		}
+		return &transformertypes.TransformerDisabledError{Err: err}
+	}
+	t.Env.AddChild(t.CNBEnv)
+	return nil
+}
+
+// GetConfig returns the transformer config
+func (t *CNBContainerizer) GetConfig() (transformertypes.Transformer, *environment.Environment) {
+	return t.Config, t.Env
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *CNBContainerizer) DirectoryDetect(dir string) (services map[string][]transformertypes.Artifact, err error) {
+	path := dir
+	cmd := environmenttypes.Command{
+		"/cnb/lifecycle/detector", "-app", t.CNBEnv.Encode(path).(string)}
+	stdout, stderr, exitcode, err := t.CNBEnv.Exec(cmd)
+	if err != nil {
+		if errors.Is(err, &environment.EnvironmentNotActiveError{}) {
+			logrus.Debugf("%s", err)
+			return nil, err
+		}
+		logrus.Errorf("Detect failed %s : %s : %d : %s", stdout, stderr, exitcode, err)
+		return nil, err
+	} else if exitcode != 0 {
+		logrus.Debugf("Detect did not succeed %s : %s : %d", stdout, stderr, exitcode)
+		return nil, nil
+	}
+	detectedServices := []transformertypes.Artifact{{
+		Paths: map[transformertypes.PathType][]string{
+			artifacts.ServiceDirPathType: {dir},
+		},
+		Configs: map[string]interface{}{artifacts.CNBMetadataConfigType: artifacts.CNBMetadataConfig{
+			CNBBuilder: t.CNBConfig.BuilderImageName,
+		}}}}
+	return map[string][]transformertypes.Artifact{"": detectedServices}, nil
+}
+
+// Transform transforms the artifacts
+func (t *CNBContainerizer) Transform(newArtifacts []transformertypes.Artifact, oldArtifacts []transformertypes.Artifact) (tPathMappings []transformertypes.PathMapping, tArtifacts []transformertypes.Artifact, err error) {
+	tArtifacts = []transformertypes.Artifact{}
+	for _, a := range newArtifacts {
+		var sConfig artifacts.ServiceConfig
+		err = a.GetConfig(artifacts.ServiceConfigType, &sConfig)
+		if err != nil {
+			logrus.Errorf("unable to load config for Transformer into %T : %s", sConfig, err)
+			continue
+		}
+		var cConfig artifacts.CNBMetadataConfig
+		err = a.GetConfig(artifacts.CNBMetadataConfigType, &cConfig)
+		if err != nil {
+			logrus.Errorf("unable to load config for Transformer into %T : %s", cConfig, err)
+			continue
+		}
+		if cConfig.ImageName == "" {
+			cConfig.ImageName = common.MakeStringContainerImageNameCompliant(sConfig.ServiceName)
+		}
+		a.Configs[artifacts.CNBMetadataConfigType] = cConfig
```

Comment:
Do we need to set it back, if imagename is not in that config?

Reply:
removed it. Not required.